core/server/src/main/java/alluxio/master/journal/ProtoBufJournalFormatter.java

```diff
@@ -40,11 +47,25 @@ public void serialize(JournalEntry entry, OutputStream outputStream) throws IOEx
   @Override
   public JournalInputStream deserialize(final InputStream inputStream) throws IOException {
     return new JournalInputStream() {
+      private final byte[] mBuffer = new byte[1024];
       private long mLatestSequenceNumber;
 
       @Override
       public JournalEntry getNextEntry() throws IOException {
-        JournalEntry entry = JournalEntry.parseDelimitedFrom(inputStream);
+        int firstByte = inputStream.read();
+        if (firstByte == -1) {
+          return null;
+        }
+        int size = CodedInputStream.readRawVarint32(firstByte, inputStream);
+        byte[] buffer = size <= mBuffer.length ? mBuffer : new byte[size];
+        int bytes = inputStream.read(buffer, 0, size);
+        if (bytes < size) {
```

Comment:
What about the case where we write only part of the corrupted journal entry? We wouldn't know where the next journal entry starts.

|15,15 byte valid entry|12,8 byte corrupted entry|10,10 byte valid entry|

the corrupted entry here happens to have 8 bytes, but we don't know that when we're trying to read it later on, so we don't know how to get to the start of the third entry